src/stackdriver-nozzle/nozzle/nozzle.go

```diff
@@ -22,45 +23,29 @@ func (e *PostContainerMetricError) Error() string {
 
 type Nozzle struct {
 	StackdriverClient stackdriver.Client
+	Serializer        serializer.Serializer
 }
 
-func (n *Nozzle) HandleEvent(eventsEnvelope *events.Envelope) error {
-	envelope := Envelope{eventsEnvelope}
-	labels := envelope.Labels()
-
-	switch envelope.GetEventType() {
-	case events.Envelope_ContainerMetric:
-		return n.postContainerMetrics(envelope)
-	case events.Envelope_ValueMetric:
-		valueMetric := envelope.GetValueMetric()
-		name := valueMetric.GetName()
-		value := valueMetric.GetValue()
-
-		err := n.StackdriverClient.PostMetric(name, value, labels)
-		return err
-	default:
-		n.StackdriverClient.PostLog(envelope, labels)
+func (n *Nozzle) HandleEvent(envelope *events.Envelope) error {
+	if n.Serializer.IsLog(envelope) {
+		log := n.Serializer.GetLog(envelope)
+		n.StackdriverClient.PostLog(log.Payload, log.Labels)
 		return nil
+	} else {
+		metrics := n.Serializer.GetMetrics(envelope)
```

Comment:
Good to point out and agree this is going to be a slow spot that will likely need some work for perf in that story